pkg/serializer/func_serializer.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2017 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package serializer
+
+var (
+	noRetry = func(i int) bool { return false }
+)
+
+type funcProp struct {
+	f        func() error
+	waitFunc func(int) bool
+}
+
+type funcSerializer struct {
+	funcs chan funcProp
+}
+
+// NewFuncSerializer returns a funcSerializer that will be used to execute
+// functions in the same order they are enqueued.
+func NewFuncSerializer(size uint) *funcSerializer {
+	return &funcSerializer{
+		funcs: make(chan funcProp, size),
+	}
+}
+
+// Run starts the funcSerializer internal worker. It will be stopped once
+// `stopCh` is closed or receives a value.
```

Comment:
This is very misleading, `stopped` indicates that this is a go routine which it is not. Why not just spawn the routine in `NewFuncSerializer`? I would also store the stop channel in the `funcSerializer` struct and provide a `Stop()` API so the channel is hidden.

Reply:
Done